### ISSUE TITLE
Fluentd parser error  

### ISSUE THREAD

**Issue by mitrakanuri** [fluent/fluentd#2792]
(https://github.com/fluent/fluentd/blob/master/CONTRIBUTING.md) first and here is the list to help us investigate the problem.

**Describe the bug**
 I am trying to parse logs being generated from Docker swarm to fluentd and strangely some of them get parsed but some do not. The logs which do not get parsed get spammed with the error below in ES which says there is a parser error .I've tried using the same log that is showing up in ES  in my local with by tailing the log using in_tail input plugin and using the same regex and it works perfectly. 
<dump an error event: error_class=Fluent::Plugin::Parser::ParserError error

```
 <source>
    @type tail
    path "/tmp/tailfile"
    pos_file "/tmp/tailfile.pos"
    tag "lab4.3"
    <parse>
      @type "none"
    </parse>
  </source>
  <filter lab4**>
    @type parser
    key_name "message"
    <parse>
      @type "regexp"
      expression ^(?<time>\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}.\d{3})\s(?<log>((.*)[\n]?)*)
    </parse>
  </filter>
  <match>
    @type stdout
  </match>
```
The same filter configuration does not work when I use it in_forward plugin even though I parsed 'log" field in the original configuration.

**To Reproduce**
Use two docker containers with fluentd and another 

**Expected behavior**
Expected the logs to show up without parser error in the logs

**Your Environment**

-Using fluentd container version 1.2.6
- Operating system: Alpine linux

If you hit the problem with older fluentd version, try latest version first.

**Your Configuration**

```
<!-- <source>
  @type forward
  port 24224
  bind 0.0.0.0
</source>

# to concatenate log events that span mulitple lines
<filter **>
  @type concat
  key log
  multiline_start_regexp ^\[\d{4}-\d{2}-\d{2}\s\d{2}:\d{2}:\d{2}\.\d+\]\
  continuous_line_regexp ^\s+
  separator "\n"
  flush_interval 3s
  timeout_label @NORMAL # handle timeout log lines the same as normal logs
</filter>

# route everything to @NORMAL label
<match **>
  @type relabel
  @label @NORMAL
</match>

<label @NORMAL>
  # declare application specific filters conf files with ONLY filters here
  @include insight.conf
 

  # send logs to Elasticsearch; container should not need to authenticate to access the ES endpoint
  <match **>
    @type copy
    <store>
      @type elasticsearch
      host "#{ENV['AWS_ELASTICSEARCH_URL']}"
      port 443
      scheme https
      ssl_version TLSv1_2
      verify_es_version_at_startup false
      default_elasticsearch_version 6

      reload_connections false
      with_transporter_log true
      @log_level debug
      request_timeout 2147483648
      logstash_format true
      logstash_dateformat %Y%m%d
      include_tag_key true
      type_name access_log
      tag_key @log_name
      flush_interval 3s
    </store>
  </match>
</label> -->
```
AND using the following for @insight.conf

```
<filter **_app-insight**>
  @type parser
  key_name log
  <parse>
   @type regexp
   expression /^(?<time>\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}.\d{3})\s(?<log>((.*)[\n]?)*)/
  </parse>
  emit_invalid_record_to_error false
</filter>

<filter **_app-insight**>
  @type record_modifier
  <record>
    logType "app"
  </record>
</filter>
```

**Your Error Log**

```
<dump an error event: error_class=Fluent::Plugin::Parser::ParserError error="pattern not match with data '2020-01-22 18:23:20.279 [apm-remote-config-poller] ERROR co.elastic.apm.agent.report.HttpUtils - Exception when closing input stream of HttpURLConnection.'" location=nil tag="xxxxx-xxxx-xxxx.xxxxtest.com_app-xxxxx_app.1.xxxxxxxx" time=1579717400 ->
```

The error log being shown is not really the error log but it is the original application log that shows upvoting in ES

**Comment by ganmacs** [comment]
Your regex may be wrong.
check it to use your actual log on https://rubular.com/

**Comment by mitrakanuri** [comment]
The regex ^(?<time>\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}.\d{3})\s(?<log>((.*)[\n]?)*) works when I test in my local machine using in_tail but it fails when I try use it along with the other docker swarm services with in_forward plugin.

I tried the https://rubular.com/ and works fine too.

**Comment by ganmacs** [comment]
it doesn't work https://rubular.com/r/SGqijMdbRdLw7Y

**Comment by mitrakanuri** [comment]
I see that you've put forward slashes there but it does work if you put the actual regex https://rubular.com/r/SGqijMdbRdLw7Y it works because the editor automatically adds the slashes.

**Comment by mitrakanuri** [comment]
Any update on this ? The current logs are being spammed with the parser error logs.

**Comment by ganmacs** [comment]
Could you give us the following information? I couldn't reproduce it now.

* Fluentd config
* The actual message which causes the error

**Comment by mitrakanuri** [comment]
@ganmacs  The fluentd config is :
```
<!-- <source>
  @type forward
  port 24224
  bind 0.0.0.0
</source>

# to concatenate log events that span mulitple lines
<filter **>
  @type concat
  key log
  multiline_start_regexp ^\[\d{4}-\d{2}-\d{2}\s\d{2}:\d{2}:\d{2}\.\d+\]\
  continuous_line_regexp ^\s+
  separator "\n"
  flush_interval 3s
  timeout_label @NORMAL # handle timeout log lines the same as normal logs
</filter>

# route everything to @NORMAL label
<match **>
  @type relabel
  @label @NORMAL
</match>

<label @NORMAL>
  # declare application specific filters conf files with ONLY filters here
  @include insight.conf
 

  # send logs to Elasticsearch; container should not need to authenticate to access the ES endpoint
  <match **>
    @type copy
    <store>
      @type elasticsearch
      host "#{ENV['AWS_ELASTICSEARCH_URL']}"
      port 443
      scheme https
      ssl_version TLSv1_2
      verify_es_version_at_startup false
      default_elasticsearch_version 6

      reload_connections false
      with_transporter_log true
      @log_level debug
      request_timeout 2147483648
      logstash_format true
      logstash_dateformat %Y%m%d
      include_tag_key true
      type_name access_log
      tag_key @log_name
      flush_interval 3s
    </store>
  </match>
</label> -->
```
And for the insight.conf the config is :

<filter **_app-insight**>
  @type parser
  key_name log
  <parse>
   @type regexp
   expression /^(?<time>\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}.\d{3})\s(?<log>((.*)[\n]?)*)/
  </parse>
  emit_invalid_record_to_error false
</filter>

<filter **_app-insight**>
  @type record_modifier
  <record>
    logType "app"
  </record>
</filter>

The log message I am receiving is :

<- dump an error event: error_class=Fluent::Plugin::Parser::ParserError error="pattern not match with data '2020-01-22 18:23:20.279 [apm-remote-config-poller] ERROR co.elastic.apm.agent.report.HttpUtils - Exception when closing input stream of HttpURLConnection.'" location=nil tag="xxxxx-xxxx-xxxx.xxxxtest.com_app-xxxxx_app.1.xxxxxxxx" time=1579717400 ->

FYI there is no error message but the logs are getting spammed with the above error message.

**Comment by ganmacs** [comment]
Please send *actual data* not error message. I can't reproduce it. and can you send us the **actual** logs **around** the error?

**Comment by mitrakanuri** [comment]
After some debugging I've noticed that the parser error spamming is being caused because the containers are producing two different log formats. I've tried the multiformat plugin to parse the logs in two different formats with no luck. Is there any way fluentd can parse two different log formats with the same tag?

<- dump an error event: error_class=Fluent::Plugin::Parser::ParserError error="pattern not match with data '2020-01-22 18:23:20.279 [apm-remote-config-poller] ERROR co.elastic.apm.agent.report.HttpUtils - Exception when closing input stream of HttpURLConnection.'" location=nil tag="xxxxx-xxxx-xxxx.xxxxtest.com_app-xxxxx_app.1.xxxxxxxx" time=1579717400 ->

The above is the data that is being sent to Elasticsearch where logs are being spammed with the parser error messages.
